main.yml

```diff
@@ -0,0 +1,24 @@
+name: main
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-yaml:
+    runs-on: ubuntu-latest
```

Comment:
is that a docker instance? 
for the script to work, the instance must have all the following packages already installed:
* vim
* python 3
* https://github.com/kislyuk/yq

Not sure about the first 2, but yq is definitely not installed by default anywhere

Reply:
It's a virtual machine, yes (though they didn't specify whether it's docker or not).
They do have some things installed (for instance they can run node and docker), but I'm not sure what is.
However, every command has it's own way in which it gets executed, depending on the action used (for instance, [yamllint](https://github.com/ibiqlik/action-yamllint) starts up a docker container, while I've also seen some use a node application).

So the `runs-on` is not really the problem (you only have `ubuntu`, `mac` and `windows` to choose from), but instead the command itself would have to start a container with the necessary libraries installed.
Not sure, how that is done correctly, yet… Maybe we do install yq etc. first on the machine? Probably the easiest, however would be to make a new project with the dockerfile. Shouldn't be too much work.

I kinda forgot about that before removing the Draft-warning ^^'

Reply:
I should probably be the one taking care of this ^^ I'll try to make some time for it next week, thanks for looking into it. 
Maybe creating a docker image and duplicate yamllint's way of doing is the simplest way to go

Reply:
I've been dockerifying a lot of things recently for my new OS, so I really don't mind doing that for this either ^^

Reply:
I've created a more generic repository for this [here](https://github.com/awesome-global-contributions/action-yq)

Reply:
Good idea, thanks for the initiative!